app/models/symphony_curr_loc_request.rb

```diff
@@ -0,0 +1,53 @@
+require 'json'
+
+# API for retrieving the current location from a barcode via Symphony Web Services
+class SymphonyCurrLocRequest
+  include ActiveModel::Model  # allows initialization with a hash of attributes
+
+  attr_accessor :barcode
+
+  def current_location
+    return '' if json.blank?
+    json['fields']['currentLocation']['key'] || ''
+  end
+
+  private
+
+  def response
+    @response ||= Faraday.get do |req|
+      req.url url
+      # need the below for symws catalog/item/barcode
+      req.headers['x-sirs-clientID'] = 'DS_CLIENT'
+      req.headers['sd-originating-app-id'] = 'requests'
+      req.headers['SD-Preferred-Role'] = 'GUEST'
+    end
+    return empty_response(@response.body) unless @response.success?
+    @response
+  rescue Faraday::Error::ConnectionFailed => e
+    empty_response(e)
+  end
+
+  def json
+    return {} unless response.success?
+    @json ||= JSON.parse(response.body)
+  rescue JSON::ParserError
+    {}
+  end
+
+  def url
+    @url ||= "#{base_url}#{barcode}?#{query_str}"
+  end
+
+  def base_url
+    @@base_url ||= "#{Settings.symphony_web_services.base_url}#{Settings.symphony_web_services.curr_loc_path}"
```

Comment:
is the idea that I would do something like
```ruby
BASE_URL = "#{Settings.symphony_web_services.base_url}#{Settings.symphony_web_services.curr_loc_path}"
```
because it will be a constant while the app is running.   Or is it suggesting I do something else?